### PR TITLE
FEATURE: directory edit

### PR DESCRIPTION
**Pull request by iTitus**

### CODE REVIEW

**Comment by iTitus** [comment]
Root:
image

**Comment by iTitus** [comment]
Child:
image

File: src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java [hunk 1]
```diff
@@ -226,6 +226,32 @@ public Directory deleteFolder(Account account, long dirId) throws MopsException
...
+            long dirId,
+            String newName,
+            DirectoryPermissions newPermissions) throws MopsException {
+        Directory directory = getDirectory(dirId);
+        securityService.checkIfRole(account, directory.getGroupOwner(), adminRole);
+
+        if (directory.getParentId() != null) {
+            if (newName == null || newName.isEmpty()) {
+                log.error("The user '{}' tried to change the name of a directory to an empty name.", account.getName());
+                throw new DatabaseException("Name leer.");
```
Comment (Segelzwerg) [inline src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java]:
```suggestion
                throw new DatabaseException("Name des Ordners darf nicht leer sein.");
```

File: src/main/java/mops/presentation/ExceptionController.java [hunk 2]
```diff
@@ -52,7 +52,11 @@ public ModelAndView resolveException(HttpServletRequest request,
     @RequestMapping("/material1/error")
     @SuppressWarnings({ "PMD.LawOfDemeter", "PMD.DataflowAnomalyAnalysis", "PMD.CyclomaticComplexity" })
     public String handleError(KeycloakAuthenticationToken token, HttpServletRequest request, Model model) {
-        Account account = Account.of(token);
+        if (token != null) {
```
Comment (Segelzwerg) [inline src/main/java/mops/presentation/ExceptionController.java]: The token should not be empty?

File: src/main/java/mops/presentation/form/EditDirectoryForm.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+     * Role Permissions.
+     */
+    private List<RolePermissionsForm> rolePermissions;
+
+    /**
+     * Creates a new EditDirectoryForm from an existing directory and permissions.
+     *
+     * @param directory   directory
+     * @param permissions directory permissions
+     * @return EditDirectoryForm
```
Comment (Segelzwerg) [inline src/main/java/mops/presentation/form/EditDirectoryForm.java]: be more specific

File: src/main/java/mops/presentation/ExceptionController.java (hunk 2 above)
Comment (iTitus) [reply src/main/java/mops/presentation/ExceptionController.java]: i had errors in my console, couldn't reproduce later tho

**Pull request merged**